interp/value.go

```diff
@@ -195,7 +195,7 @@ func genValue(n *node) func(*frame) reflect.Value {
 		}
 		if n.sym != nil {
 			i := n.sym.index
-			if i < 0 {
+			if i < 0 && n != n.sym.node {
```

Comment:
Is this enough to also protect us in case of indirect recursion?
for example:
n != n.sym.node, but n == n.sym.node.sym.node ?